### PR TITLE
Bug/contiguous times

### PR DESCRIPTION
**Pull request by markysoft**
Combine contiguous opening times into a single session

### CODE REVIEW

File: app/lib/joinContiguousTimes.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      o.prev = session;
+    }
+    /* eslint-enable no-param-reassign */
+    return o;
+  }, { list: [], prev: undefined });
+
+  return fixedKeySessions.list;
+}
+
+function joinTimes(type) {
```
Comment (st3v3nhunt) [inline app/lib/joinContiguousTimes.js]: IMO, `times` is a better name than `type`.

File: test/unit/joinContiguousTimes.js [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+          ]
+        }
+      }
+    };
+
+    joinContiguousTimes(pharmacy);
+    expect(pharmacy.openingTimes.general.monday.length).to.equal(2);
+    expect(pharmacy.openingTimes.general.monday[0].opens).to.equal('09:00');
+    expect(pharmacy.openingTimes.general.monday[0].closes).to.equal('13:00');
+    expect(pharmacy.openingTimes.general.monday.length).to.equal(2);
```
Comment (st3v3nhunt) [inline test/unit/joinContiguousTimes.js]: Expectation already run https://github.com/nhsuk/pharmacy-data-etl/pull/37/files#diff-41ae383597d8061bc576d4aaa5862d8bR178

File: test/unit/joinContiguousTimes.js [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    expect(pharmacy.openingTimes.general.monday.length).to.equal(1);
+    expect(pharmacy.openingTimes.general.monday[0].opens).to.equal('09:00');
+    expect(pharmacy.openingTimes.general.monday[0].closes).to.equal('19:30');
+    expect(pharmacy.openingTimes.general.tuesday.length).to.equal(1);
+    expect(pharmacy.openingTimes.general.tuesday[0].opens).to.equal('09:01');
+    expect(pharmacy.openingTimes.general.tuesday[0].closes).to.equal('19:31');
+    expect(pharmacy.openingTimes.general.wednesday.length).to.equal(1);
+    expect(pharmacy.openingTimes.general.wednesday[0].opens).to.equal('09:02');
+    expect(pharmacy.openingTimes.general.wednesday[0].closes).to.equal('19:32');
+    expect(pharmacy.openingTimes.general.tuesday[0].closes).to.equal('19:31');
```
Comment (st3v3nhunt) [inline test/unit/joinContiguousTimes.js]: Duplicate expectation https://github.com/nhsuk/pharmacy-data-etl/pull/37/files#diff-41ae383597d8061bc576d4aaa5862d8bR136

File: test/unit/joinContiguousTimes.js [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    expect(pharmacy.openingTimes.general.thursday.length).to.equal(1);
+    expect(pharmacy.openingTimes.general.thursday[0].opens).to.equal('09:03');
+    expect(pharmacy.openingTimes.general.thursday[0].closes).to.equal('19:33');
+    expect(pharmacy.openingTimes.general.friday.length).to.equal(1);
+    expect(pharmacy.openingTimes.general.friday[0].opens).to.equal('09:04');
+    expect(pharmacy.openingTimes.general.friday[0].closes).to.equal('19:34');
+    expect(pharmacy.openingTimes.general.saturday.length).to.equal(1);
+    expect(pharmacy.openingTimes.general.saturday[0].opens).to.equal('09:05');
+    expect(pharmacy.openingTimes.general.saturday[0].closes).to.equal('19:35');
+    expect(pharmacy.openingTimes.general.friday[0].closes).to.equal('19:34');
```
Comment (st3v3nhunt) [inline test/unit/joinContiguousTimes.js]: Duplicate expectation https://github.com/nhsuk/pharmacy-data-etl/pull/37/files#diff-41ae383597d8061bc576d4aaa5862d8bR146

File: test/unit/joinContiguousTimes.js [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+          ]
+        }
+      }
+    };
+
+    joinContiguousTimes(pharmacy);
+    expect(pharmacy.openingTimes.general.monday.length).to.equal(2);
+    expect(pharmacy.openingTimes.general.monday[0].opens).to.equal('09:00');
+    expect(pharmacy.openingTimes.general.monday[0].closes).to.equal('13:00');
+    expect(pharmacy.openingTimes.general.monday.length).to.equal(2);
```
Comment (st3v3nhunt) [inline test/unit/joinContiguousTimes.js]: Duplicate expectation https://github.com/nhsuk/pharmacy-data-etl/pull/37/files#diff-41ae383597d8061bc576d4aaa5862d8bR226

File: test/integration/azureService.js [hunk 6]
```diff
@@ -5,7 +5,7 @@ const azureService = require('../../app/lib/azureService');
 const expect = chai.expect;
 // writing to a shared area, timestamp file to avoid collisions
 const name = `${new Date().getTime()}test.json`;
-const timeout = 5000;
+const timeout = 15000;
```
Comment (st3v3nhunt) [inline test/integration/azureService.js]: Does this need increasing to `15000` or did it fail as a one off and could it be left at `5000`?

File: .snyk [hunk 7]
```diff
@@ -13,4 +13,15 @@ ignore:
     - node-pre-gyp > tar-pack > debug:
         reason: Not user facing
         expires: '2017-11-24T08:35:43.242Z'
-patch: {}
+  'npm:shelljs:20140723':
+    - eslint-config-nhsuk > eslint-plugin-json > jshint > shelljs:
+        reason: no patch available
+        expires: '2018-02-02T14:12:43.543Z'
+# patches apply the minimum changes required to fix a vulnerability
+patch:
```
Comment (st3v3nhunt) [inline .snyk]: I didn't think patching actually did anything. Does the data on the ignores above (in the file) need updating and then the patching section can be removed?

File: test/integration/azureService.js (hunk 6 above)
Comment (markysoft) [reply test/integration/azureService.js]: I was getting a lot of failures locally due to poor network speeds in the office during lunch time until I bumped the timeout up. This is with a network cable, not wifi.

File: .snyk (hunk 7 above)
Comment (markysoft) [reply .snyk]: Deleting the `.snyk` file then re-running the wizard has removed the ignores

**Pull request merged**